ocamltest/ocamltest_stdlib.ml

```diff
@@ -84,22 +84,29 @@ end
 module Sys = struct
   include Sys
 
-  let run_system_command prog args =
-    let command = Filename.quote_command prog args in
-    match Sys.command command with
-    | 0 -> ()
-    | _ as exitcode ->
-      Printf.eprintf "System command %s failed with status %d\n%!"
-        command exitcode;
-      exit 3
-
-  let mkdir dir =
-    if not (Sys.file_exists dir) then
-      run_system_command "mkdir" [dir]
+  let erase_file path =
+    try Sys.remove path
+    with Sys_error _ when Sys.win32 && Ocamltest_config.libunix <> None ->
+      (* Deal with read-only attribute on Windows. *)
+      Unix.chmod path 0o666;
+      Sys.remove path
+
+  let rm_rf path =
+  let rec erase path =
+    if Sys.is_directory path
+    then Array.iter (fun entry -> erase (Filename.concat path entry))
+                    (Sys.readdir path)
+    else erase_file path
+  in
+    if Sys.file_exists path then erase path
 
   let rec make_directory dir =
     if Sys.file_exists dir then ()
-    else (make_directory (Filename.dirname dir); mkdir dir)
+    else let () = make_directory (Filename.dirname dir) in
+         if not (Sys.file_exists dir) then
+           try Sys.mkdir dir 0o777
+           with Sys_error _ -> exit 3
```

Comment:
You can't just "exit" without any explanation!  In the previous code, the external "mkdir -p" command would have produced a detailed error message.  Suggested change:
```suggestion
           try Sys.mkdir dir 0o777
           with Sys_error msg ->
             Printf.eprintf "Cannot create directory: %s\n" msg; exit 3
```
Alternative: don't catch the exception here, let it propagate up, and have a handler for all Sys_error exceptions that print the message and exit 3.

Reply:
Oops, sorry - I'd said I'd put one in and then didn't! Fix coming up...